tests/Unit/Evolution/Systems/RelativisticEuler/Valencia/Test_Characteristics.cpp

```diff
@@ -0,0 +1,92 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <array>
+#include <cmath>
+#include <cstddef>
+#include <random>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/Direction.hpp"
+#include "Evolution/Systems/RelativisticEuler/Valencia/Characteristics.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/StdHelpers.hpp"
+#include "tests/Unit/Domain/DomainTestHelpers.hpp"
+#include "tests/Unit/Pypp/CheckWithRandomValues.hpp"
+#include "tests/Unit/Pypp/Pypp.hpp"
+#include "tests/Unit/Pypp/SetupLocalPythonEnvironment.hpp"
+#include "tests/Utilities/MakeWithRandomValues.hpp"
+
+// IWYU pragma: no_forward_declare Tensor
+
+namespace {
+
+template <size_t Dim>
+void test_characteristic_speeds(const DataVector& used_for_size) noexcept {
+  //  Arbitrary random numbers can produce a negative radicand in Lambda^\pm.
+  //  This bound helps to prevent that situation.
+  const double max_value = 1.0 / sqrt(Dim);
+  pypp::check_with_random_values<6>(
+      &RelativisticEuler::Valencia::characteristic_speeds<Dim>, "TestFunctions",
+      "characteristic_speeds",
+      {{{0.0, 1.0},
+        {-1.0, 1.0},
+        {-max_value, max_value},
+        {0.0, 1.0},
+        {0.1, 1.0},
```

Comment:
Is there a problem with small sound speeds?

Reply:
Definitely not. I'll extend the range down to 0.